src/test/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/ProcedureRequestMapperTest.java

```diff
@@ -1,20 +1,278 @@
+            message, null, null, null, null, null);
+
+        assertThat(procedureRequest).isNotEmpty();
+        assertThat(procedureRequest.get().getNote()).hasSize(3)
+            .extracting(Annotation::getText)
+            .containsExactly("COELIAC", "JAUNDICE  ?OBSTRUCTIVE", "GASTRIC ULCER DECLINE");
+    }
+
+    @Test
+    void testMapToProcedureRequestStatus() {
+        final Message message = new Message(List.of(
+            "S02+02", // ServiceReportDetails
+            "S06+06", // InvestigationSubject
+            "S10+10",
+            "CIN+UN",
+            "FTX+CID+++COELIAC"
+        ));
+
+        final Optional<ProcedureRequest> procedureRequest = procedureRequestMapper.mapToProcedureRequest(
+            message, null, null, null, null, null);
+
+        assertThat(procedureRequest).isNotEmpty();
+        assertThat(procedureRequest.get().getStatus().toCode()).isEqualTo("unknown");
+    }
+
+    @Test
+    void testMapToProcedureRequestUndefinedStatus() {
+        final Message message = new Message(List.of(
+            "S02+02", // ServiceReportDetails
+            "S06+06", // InvestigationSubject
+            "S10+10",
+            "CIN+Undefined Status",
+            "FTX+CID+++COELIAC"
+        ));
+
+        assertThatThrownBy(() -> procedureRequestMapper.mapToProcedureRequest(
+            message, null, null, null, null, null))
+            .isInstanceOf(IllegalArgumentException.class)
+            .hasMessage("No Report Status Code for 'Undefined Status'");
+    }
+
+    @Test
+    void testMapToProcedureRequestIntent() {
+        final Message message = new Message(List.of(
+            "S02+02", // ServiceReportDetails
+            "S06+06", // InvestigationSubject
+            "S10+10",
+            "CIN+UN",
+            "FTX+CID+++COELIAC"
+        ));
+
+        final Optional<ProcedureRequest> procedureRequest = procedureRequestMapper.mapToProcedureRequest(
+            message, null, null, null, null, null);
+
+        assertThat(procedureRequest).isNotEmpty();
+        assertThat(procedureRequest.get().getIntent()).isEqualTo(ProcedureRequestIntent.NULL);
+    }
+
+    @Test
+    void testMapToProcedureRequestPatientReference() {
+        final Message message = new Message(List.of(
+            "S02+02", // ServiceReportDetails
+            "S06+06", // InvestigationSubject
+            "S10+10",
+            "CIN+UN",
+            "FTX+CID+++COELIAC"
+        ));
+        when(fullUrlGenerator.generate(any(Patient.class))).thenReturn("patient-full-url");
+
+        final Optional<ProcedureRequest> procedureRequest = procedureRequestMapper.mapToProcedureRequest(
+            message, mock(Patient.class), null, null, null, null);
+
+        assertThat(procedureRequest).isNotEmpty();
+        assertThat(procedureRequest)
+            .satisfies(subject -> assertThat(subject.get().getSubject()).extracting(Reference::getReference)
+                .isEqualTo("patient-full-url"));
```

Comment:
Unfortunately this doesn't solve the problem of calling `.get()` without `isPresent()` check - could be due to the nested `assertThat` check. You could try something like this which includes the `isNotEmpty`:

```suggestion
        assertThat(procedureRequest)
            .isNotEmpty()
            .hasValueSatisfying(procedure ->
                assertThat(procedure.getSubject().getReference()).isEqualTo("patient-full-url"));
```